docs/sql.md

```diff
@@ -0,0 +1,120 @@
+# Druid SQL
+
+Scruid provides querying via [Druid SQL](https://druid.apache.org/docs/latest/querying/sql.html) by using the 
+processed string function `sql`. 
+
+```scala
+import scala.concurrent.Future
+import ing.wbaa.druid.SQL._
+import ing.wbaa.druid.DruidSQLResults
+
+val query = sql"""SELECT COUNT(*) as "count" FROM wikipedia WHERE "__time" >= TIMESTAMP '2015-09-12 00:00:00'"""
+
+val response: Future[DruidSQLResults] = query.execute()
+
+// The resulting response can also mapped to a case class
+case class Result(count: Double)
+
+val result: Future[Result] = response.map(_.list[Result])
+```
+
+Function `sql`, allows multiline queries:
+
+```scala
+import ing.wbaa.druid.SQL._
+
+val query = sql"""
+    |SELECT COUNT(*) as "count" 
+    |FROM wikipedia 
+    |WHERE "__time" >= TIMESTAMP '2015-09-12 00:00:00'
+    """.stripMargin
+```
+
+Supports standard string interpolation:
```

Comment:
Bit of a philosophical question: _should_ we support this? 

Given that "injections" take the number 1 spot in the [OWASP top-10](https://owasp.org/www-project-top-ten/). Given that we have parameterized querying as described on line 49 and onwards. I think it would be more responsible to just not offer standard string interpolation, thereby making it harder to do the wrong thing?

People can still create the problem in other ways (e.g. using `+`), that's not the point. My idea is that maybe we shouldn't pave this road.

Alternatively, it'd be very neat to have our own string interpolator that results in a parameterized query and that way get the best of both worlds. If that's difficult, we could do one that only accepts safe types such as `Int` and raises a compiler error if one attempts to interpolate an unsafe type. However, any such thing is more ambitious than we need to be, and would delay this PR, so I would not go for a custom interpolator at all.

Reply:
The standard string interpolation is provided for simplicity (e.g., quick queries in a notebook). 

In production systems, to avoid any "sql injections" you can switch to parameterized queries, which are also supported in this PR. 

Like the example below:

```scala
import java.time.LocalDateTime
import ing.wbaa.druid.SQLQuery

import ing.wbaa.druid.SQL._

val fromDateTime  = LocalDateTime.of(2015, 9, 12, 0, 0, 0, 0)
val untilDateTime = fromDateTime.plusDays(1)

val queryParameterized: SQLQuery.Parameterized =
  sql"""
  |SELECT FLOOR(__time to HOUR) AS hourTime, count(*) AS "count"
  |FROM wikipedia
  |WHERE "__time" BETWEEN ? AND ?
  |GROUP BY 1
  |""".stripMargin.parameterized

val query: SQLQuery = queryParameterized
  .withParameter(fromDateTime)
  .withParameter(untilDateTime)
  .create()
```

In practice Scruid, is offering both solutions. 

If we would like to enforce offering only the second option (i.e., parameterized queries) I can remove the string interpolation functionality.

Reply:
I think i'm in favour of removing the string interpolation indeed

Reply:
I will try to create an interpolator that constructs parameterized queries underneath